src/core/json_object.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <memory>
+#include <memory_resource>
+#include <optional>
+#include <string_view>
+
+namespace jsoncons {
+struct sorted_policy;
+template <typename CharT, typename Policy, typename Allocator> class basic_json;
+}  // namespace jsoncons
+
+// the last one in object.h is OBJ_STREAM and it is 6,
+// this will add enough place for Redis types to grow
+constexpr unsigned OBJ_JSON = 15;
+
+namespace dfly {
+
+namespace detail {
+
+// This is temporary, there is an issue right now with jsoncons about using jsonpath
+// with custom allocator. once it would resolved, we would change this to use custom allocator
+// that allocate memory from mimalloc
+using JsonType = jsoncons::basic_json<char, jsoncons::sorted_policy, std::allocator<char>>;
+
+class JsonWrapper {
+ public:
+  using InternalObj = JsonType*;
+
+  JsonWrapper() = default;
+  JsonWrapper(JsonWrapper&&) = default;
+  JsonWrapper& operator=(JsonWrapper&&) = default;
+
+  explicit JsonWrapper(JsonType&& j, std::pmr::memory_resource* mr);
```

Comment:
I would require that the allocation be done from the outside. Since this object as really "just" holding a pointer that it manged, I think it would be best to have it doing the actual memory allocation for the internal object as well as the deallocation. Without it it should be manage from the outside, which would make this object a little bit "redundant"
What this object really does is managing the lifetime of the handler we have in the union at Compact Object

Reply:
The allocation will still be outside since you pass JsonType that you create outside. When the json bug will be fixed you will need to pass custom allocator outside as well. This is why I think this interface is broken anyway. 
what is the `sizeof(JsonType)` ?